docs/design/2018-10-24-view-support.md

```diff
@@ -0,0 +1,49 @@
+# Proposal: Implement View Feature
+
+- Author(s):     AndrewDi
+- Last updated:  2018-10-24
+- Discussion at: https://github.com/pingcap/tidb/issues/7974
+
+## Abstract
+This proposal proposes to implement view feature in TiDB, aimed to make SQL more easy to write.
+
+## Background
+A database view is a searchable object in a database that is defined by a query. Though a view doesn’t store data, some refer to a view as “virtual tables,” you can query a view like you can query a table. A view can combine data from two or more tables, using joins, and also just contain a subset of information. This makes them convenient to abstract, or hide, complicated queries.
+
+Below is a visual depiction of a view:
+  ![AnatomyOfAview](.imgs/view.png)
+  
+A view is created from a query using the CREATE OR REPLACE VIEW command. In the example below we are creating a PopularBooks view based on a query which selects all Books that have the IsPopular field checked. Following is the query:
+`CREATE OR REPLACE VIEW PopularBooks AS SELECT ISBN, Title, Author,PublishDate FROM Books WHERE IsPopular=1`  
+
+Once a view is created, you can use it as you query any table in a SELECT statement. For example, to list all the popular book titles ordered by an author, you could write:  
+`SELECT Author, Title FROM PopularBooks ORDER BY Author`  
+
+In general you can use any of the SELECT clauses, such as GROUP BY, in a select statement containing a view.
+
+## Proposal
+I implement the `create or replace` command to create a view with the SELECT statement. Currently, I reuse `drop table` to drop a view, and `show full tables` can also display the view list with different table types. The `show create table` command can also regenerate a view create command.
+
+## Rationale
+View is just a tableinfo object store with select statement, view only support two types ddl operation, which are `CREATE` and `DROP`. Currently, view can only used within `SELECT` statement, I do not implement a updateable view.
+
+## Compatibility
+This proposal has no effect on the compatibility.
```

Comment:
It makes tidb more compatible with MySQL.

Reply:
Aha, very well.